benchalerts/benchalerts/pipeline_steps/conbench.py

```diff
@@ -121,3 +125,80 @@ def _get_one_run_comparison(self, run_id: str) -> RunComparisonInfo:
                 )
 
         return run_comparison
+
+
+class GetConbenchZComparisonStep(GetConbenchZComparisonForRunsStep):
+    """An ``AlertPipeline`` step to get information from Conbench comparing the runs on
+    a contender commit to their baselines, using a z-score threshold. This is always the
+    first step of the pipeline.
+
+    Parameters
+    ----------
+    commit_hash
+        The commit hash of the contender commit to compare. Needs to match EXACTLY what
+        Conbench has stored; typically 40 characters. It can't be a shortened version of
+        the hash.
+    baseline_run_type
+        The type of baseline to use. See ``BaselineCandidates`` for options.
+    z_score_threshold
+        The (positive) z-score threshold to send to the Conbench compare endpoint.
+        Benchmarks with a z-score more extreme than this threshold will be marked as
+        regressions or improvements in the result. Default is to use whatever Conbench
+        uses for default (at the time of writing, this is 5).
+    conbench_client
+        A ConbenchClient instance. If not given, one will be created using the standard
+        environment variables.
+    step_name
+        The name for this step. If not given, will default to this class's name.
+
+    Returns
+    -------
+    FullComparisonInfo
+        Information about each run associated with the contender commit, and a
+        comparison to its baseline run if that exists.
+
+    Notes
+    -----
+    Environment variables
+    ~~~~~~~~~~~~~~~~~~~~~
+    ``CONBENCH_URL``
+        The URL of the Conbench server. Only required if ``conbench_client`` is not
+        provided.
+    ``CONBENCH_EMAIL``
+        The email to use for Conbench login. Only required if ``conbench_client`` is not
+        provided and the server is private.
+    ``CONBENCH_PASSWORD``
+        The password to use for Conbench login. Only required if ``conbench_client`` is
+        not provided and the server is private.
+    """
+
+    def __init__(
+        self,
+        commit_hash: str,
+        baseline_run_type: BaselineRunCandidates,
+        z_score_threshold: Optional[float] = None,
+        conbench_client: Optional[LegacyConbenchClient] = None,
```

Comment:
What do we think about using the new one (`ConbenchClient` backed by `RetryingHTTPClient`) with a max-retry-time adjusted to the context this here runs in (like: retry for 90 seconds or so)?

Reply:
I like this idea but let's do it in a followup PR. I'll make an issue.

What will make this take a while is that this library's approach to mocking a Conbench server in unit tests makes heavy use of the requests `adapter` paradigm, which the new client stopped using. So unfortunately it will be a little more than a one-line change.

Reply:
#1186